src/main/java/org/opensearch/flowframework/workflow/AbstractRegisterLocalModelStep.java

```diff
@@ -90,68 +84,65 @@ public CompletableFuture<WorkflowData> execute(
             String modelName = (String) inputs.get(NAME_FIELD);
             String modelVersion = (String) inputs.get(VERSION_FIELD);
-            String description = (String) inputs.get(DESCRIPTION_FIELD);
-            MLModelFormat modelFormat = MLModelFormat.from((String) inputs.get(MODEL_FORMAT));
-            String modelGroupId = (String) inputs.get(MODEL_GROUP_ID);
+            String modelFormat = (String) inputs.get(MODEL_FORMAT);
+
+            // Extract non-common fields
+            String functionName = (String) inputs.get(FUNCTION_NAME);
             String modelContentHashValue = (String) inputs.get(MODEL_CONTENT_HASH_VALUE);
+            String url = (String) inputs.get(URL);
             String modelType = (String) inputs.get(MODEL_TYPE);
             String embeddingDimension = (String) inputs.get(EMBEDDING_DIMENSION);
-            FrameworkType frameworkType = FrameworkType.from((String) inputs.get(FRAMEWORK_TYPE));
+            String frameworkType = (String) inputs.get(FRAMEWORK_TYPE);
+
+            // Extract optional fields
+            String description = (String) inputs.get(DESCRIPTION_FIELD);
+            String modelGroupId = (String) inputs.get(MODEL_GROUP_ID);
             String allConfig = (String) inputs.get(ALL_CONFIG);
-            String url = (String) inputs.get(URL);
-            String functionName = (String) inputs.get(FUNCTION_NAME);
             final Boolean deploy = (Boolean) inputs.get(DEPLOY_FIELD);
 
-            // Create Model configuration
-            TextEmbeddingModelConfigBuilder modelConfigBuilder = TextEmbeddingModelConfig.builder()
-                .modelType(modelType)
-                .embeddingDimension(Integer.valueOf(embeddingDimension))
-                .frameworkType(frameworkType);
-            if (allConfig != null) {
-                modelConfigBuilder.allConfig(allConfig);
-            }
-            MLModelConfig modelConfig = modelConfigBuilder.build();
-
-            // Create register local model input
+            // Build register model input
             MLRegisterModelInputBuilder mlInputBuilder = MLRegisterModelInput.builder()
                 .modelName(modelName)
                 .version(modelVersion)
-                .modelFormat(modelFormat)
-                .hashValue(modelContentHashValue)
-                .modelConfig(modelConfig)
-                .url(url);
+                .modelFormat(MLModelFormat.from(modelFormat));
+
+            if (functionName != null) {
+                mlInputBuilder.functionName(FunctionName.from(functionName));
+            }
+            if (modelContentHashValue != null) {
+                mlInputBuilder.hashValue(modelContentHashValue);
+            }
+            if (url != null) {
+                mlInputBuilder.url(url);
+            }
+            if (Stream.of(modelType, embeddingDimension, frameworkType).allMatch(x -> x != null)) {
```

Comment:
Are these the required keys or can be optional?

Reply:
They can be optional or required depending on the implementing sub-class. In this case, these fields are only applicable for local custom model registration. We check for all the possible keys here though in this abstract class, if they're null, then its essentially no-op